bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpTransformer.java

```diff
@@ -127,32 +184,90 @@ private AttributesImpl getLinkAttributes(String pagePath) {
         return attributes;
     }
 
-    @Override
-    public void setContentHandler(ContentHandler contentHandler) {
-        this.contentHandler = contentHandler;
+    /**
+     * Aggregates the AMP specific component js.
+     * @param request The request used to resolve the page and its resources.
+     * @return The aggregated AMP component js.
+     */
+    private String getJsContent(SlingHttpServletRequest request) {
+
+        if (StringUtils.isBlank(cfg.getHeadlibName())) {
+            LOG.error("Headlib name not defined. Failed to aggregate AMP component js.");
+            return "";
+        }
+
+        StringBuilder output = new StringBuilder();
+
+        // Retrieve a set of the component's resource types.
+        Set<String> resourceTypes =
+            Utils.getResourceTypes(request.getResource(), cfg.getHeadlibResourceTypeRegex(), new HashSet<>());
+
+        try (ResourceResolver resourceResolver = resolverFactory.getServiceResourceResolver(
```

Comment:
you could turn this into a util (since it's already used in the ClientLibraryAggregatorServiceImpl)